File: Programas/NoInformados/Rompecabezas/2. Busqueda en Amplitud/PYTHON BFS/Principal.py
from Rompecabezas import Rompecabezas
from BFS import BFS

print( "Proporcione el Tablero:" )
renglon1 = list(map( int , input().split() ))
renglon2 = list(map( int , input().split() ))
renglon3 = list(map( int , input().split() ))

nodoInicial = Rompecabezas()
nodoInicial.modificarCeldas( renglon1+renglon2+renglon3 )

print( "Nodo Inicial: " )
print( nodoInicial )

ruta = BFS.Buscar( nodoInicial , Rompecabezas() )

print( "Ruta de Solucion: " )
for nodo in ruta:
   print( nodo )

File: Programas/NoInformados/Rompecabezas/2. Busqueda en Amplitud/C++ BFS/Principal.cpp
#include "./Header/Rompecabezas.h"
#include "./Header/BFS.h"
#include<vector>

int main(){

    int tablero[9];
    std::cout << "Proporcione el tablero:" << "\n";
    for( int i = 0 ; i < 9 ; i++ ){
        std::cin >> tablero[i];
    }

    Rompecabezas * n1 = new Rompecabezas( nullptr , ' ' , 0 );
    n1->modificarCeldas(tablero);

    std::cout << "Nodo Inicial" << "\n";
    n1->mostrarRompecabezas();

    std::vector<Rompecabezas*> ruta = BFS::Bucar( n1 , new Rompecabezas( nullptr , ' ' , 0 ) );
    
    std::cout << "Ruta" << "\n";

    for( int i = ruta.size()-1 ; i >= 0 ; i-- ){
        ruta[i]->mostrarRompecabezas();
    }

}